### PR TITLE
feat(divider): implement component

### PR DESCRIPTION
**Pull request by skhamvon**
Refactor component Divider
- contrasted, separator & size attributes removed
- new attribute spacing
- always show a visible separator now

### CODE REVIEW

**Review by dpellier — CHANGES_REQUESTED** [review]

File: packages/storybook/stories/components/divider/divider.stories.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  render: () => html`
+<ods-divider></ods-divider>
+  `,
+};
+
+
+export const Color: StoryObj = {
+  tags: ['isHidden'],
+  render: () => html`
+<ods-divider color="light"></ods-divider>
```
Comment (dpellier) [inline packages/storybook/stories/components/divider/divider.stories.ts:76]: please add spacing on each, otherwise the rendering is not easy to see

**Review by dpellier — CHANGES_REQUESTED** [review]

**Review by feoche — CHANGES_REQUESTED** [review]
### Storybook
- [x] Component should be in "Layout" section

### Documentation
- [ ] `ODS_DIVIDER_SPACING` is not ordered
- [x] Add `spacing="16"` on the two "Color" examples

### Migration
- [x] Add `spacing` attribute change, values have been renamed (see enum)
- [x] Typo "seperator" → "separator"
- [x] "Divider is always shown as visible line now." → "Divider is always shown by default"

### Demo
- [x] Update `customCss` description to have `background-color` instead of `color`

File: packages/ods/src/components/divider/src/components/ods-divider/ods-divider.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import { ODS_DIVIDER_SPACING } from '../../constants/divider-spacing';
+
+@Component({
+  shadow: true,
+  styleUrl: 'ods-divider.scss',
+  tag: 'ods-divider',
+})
+export class OdsDivider {
+  @Prop({ reflect: true }) color: OdsDividerColor = ODS_DIVIDER_COLOR.light;
+  @Prop({ reflect: true }) spacing: OdsDividerSpacing = ODS_DIVIDER_SPACING._2;
```
[the remark this answers is not in the document]
Comment (skhamvon) [reply packages/ods/src/components/divider/src/components/ods-divider/ods-divider.tsx]: It's added

File: packages/ods/src/components/divider/src/components/ods-divider/ods-divider.scss [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      background-color: var(--ods-color-neutral-100);
+    }
+
+    &--dark {
+      background-color: var(--ods-color-neutral-700);
+    }
+
+    @each $spacing in $spacings {
+      &--#{$spacing} {
+        margin-top: calc(#{$spacing}px / 2);
```
[the remark this answers is not in the document]
Comment (skhamvon) [reply packages/ods/src/components/divider/src/components/ods-divider/ods-divider.scss]: Thanks, It's updated

File: packages/storybook/stories/components/divider/divider.stories.ts (hunk 1 above)
Comment (skhamvon) [reply packages/storybook/stories/components/divider/divider.stories.ts]: Done

File: packages/storybook/stories/components/divider/migration.from.17.x.mdx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+The 2 possible values are now `light` and `dark`.
+
+`contrasted ` <img src="https://img.shields.io/badge/removed-FF0000" />
+
+Has been removed.
+
+Color attribute can be use with `dark` for contrasted content.
+
+`seperator ` <img src="https://img.shields.io/badge/removed-FF0000" />
```
Comment (Leotheluck) [inline packages/storybook/stories/components/divider/migration.from.17.x.mdx:23]: typo `separator`

File: packages/storybook/stories/components/divider/migration.from.17.x.mdx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+<osds-divider contrasted></osds-divider>
+
+<!-- is now -->
+
+<osds-divider color="dark"></osds-divider>
+```
+
+Separator:
+```html
+<osds-divider seperator="false"></osds-divider>
```
Comment (Leotheluck) [inline packages/storybook/stories/components/divider/migration.from.17.x.mdx:64]: typo `separator`

File: packages/storybook/stories/components/divider/migration.from.17.x.mdx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+<style>
+  .custom {
+    background-color: transparent;
+  }
+</style>
+```
+
+
+Size:
```
Comment (Leotheluck) [inline packages/storybook/stories/components/divider/migration.from.17.x.mdx:78]: Size 1 is now equivalent to spacing _2 - Spacing attribute works with pixel values

File: packages/storybook/stories/components/divider/migration.from.17.x.mdx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+Contrasted:
+```html
+<osds-divider contrasted></osds-divider>
+
+<!-- is now -->
+
+<osds-divider color="dark"></osds-divider>
+```
+
+Seperator:
```
[the remark this answers is not in the document]
Comment (skhamvon) [reply packages/storybook/stories/components/divider/migration.from.17.x.mdx]: Typo fixed

File: packages/storybook/stories/components/divider/migration.from.17.x.mdx [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+Color attribute can be use with `dark` for contrasted content.
+
+`seperator ` <img src="https://img.shields.io/badge/removed-FF0000" />
+
+Has been removed.
+
+Divider is always shown by default.
+
+`size ` <img src="https://img.shields.io/badge/removed-FF0000" />
+
```
Comment (Leotheluck) [inline packages/storybook/stories/components/divider/migration.from.17.x.mdx:30]: Documentation isn't clear regarding the new attribute, please use the format of the other migration guides to announce the new attribute & specify that Spacing works with `px` values.

**Review by Leotheluck — CHANGES_REQUESTED** [review]
Add a global comment in the migration guide to specify that the divider is a visual separator and shouldn't be used as a vertical separating measure (check button's migration guide `usage changes` to write this)

File: packages/storybook/stories/components/divider/migration.from.17.x.mdx (hunk 4 above)
Comment (skhamvon) [reply packages/storybook/stories/components/divider/migration.from.17.x.mdx]: Fixed

File: packages/storybook/stories/components/divider/migration.from.17.x.mdx (hunk 5 above)
Comment (skhamvon) [reply packages/storybook/stories/components/divider/migration.from.17.x.mdx]: Fixed

File: packages/storybook/stories/components/divider/migration.from.17.x.mdx (hunk 6 above)
Comment (skhamvon) [reply packages/storybook/stories/components/divider/migration.from.17.x.mdx:79]: Fixed

File: packages/storybook/stories/components/divider/migration.from.17.x.mdx (hunk 8 above)
Comment (skhamvon) [reply packages/storybook/stories/components/divider/migration.from.17.x.mdx:30]: Description of spacing has been changed

File: packages/storybook/stories/components/divider/documentation.mdx [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+# Overview
+
+The divider is a spacer dedicated to represent the empty space. it displays a simple line separator with different spacing and colors.
+
+<Canvas of={ DividerStories.Default } sourceState='none' />
+
+<DocNavigator of={ DividerStories } />
+
+# API
```
Comment (dpellier) [inline packages/storybook/stories/components/divider/documentation.mdx:19]: Remove this line, not needed with latest doc script change

**Review by dpellier — CHANGES_REQUESTED** [review]

**Review by dpellier — APPROVED** [review]

**Review by Leotheluck — APPROVED** [review]

**Pull request merged**